src/NBitcoin/ChainedBlock.cs

```diff
@@ -449,13 +452,51 @@ public ChainedBlock GetAncestor(int height)
 
             ChainedBlock current = this;
 
-            while (true)
+            int currentHeight = this.Height;
+            while (currentHeight > height)
             {
-                if (current.Height == height)
-                    return current;
-
-                current = current.Previous;
+                int heightSkip = GetSkipHeight(currentHeight);
+                int heightSkipPrev = GetSkipHeight(currentHeight - 1);
+                if ((current.Skip != null) && ((heightSkip == height) || ((heightSkip > height) && !((heightSkipPrev < (heightSkip - 2)) && (heightSkipPrev >= height)))))
+                {
+                    // Only follow Skip if Previous.Skip isn't better than Skip.Previous.
+                    current = current.Skip;
+                    currentHeight = heightSkip;
+                }
+                else
+                {
+                    current = current.Previous;
+                    currentHeight--;
+                }
             }
+
+            return current;
+        }
+
+        /// <summary>
+        /// Compute what height to jump back to for the skip block given this height.
+        /// </summary>
+        /// <param name="height">Height to compute skip height for.</param>
+        /// <returns>The height to skip to.</returns>
+        private int GetSkipHeight(int height)
+        {
+            if (height < 2)
+                return 0;
+
+            // Determine which height to jump back to. Any number strictly lower than height is acceptable,
+            // but the following expression was taken from bitcoin core. There is was tested in simulations
+            // and performed well (max 110 steps to go back up to 2^18 blocks).
```

Comment:
So basically the jumps are of 110 blocks? (or up to 110 blocks?) will there be incremental steps?
Maybe a bit more explanation unless it is THAT simple.

Reply:
That comment was almost directly from core. I don't really understand it myself but I was hoping it was just me :grinning: I guess not. I will look at the code more carefully and see if I can write a cleaner comment.

Reply:
no, jumps are not 110 blocks :D jumps are exponential and 110 hops is expected number of hops when the chain height is 2^18, which is 262144. So traversing whole chain using `.Previous`  would take 262144 steps, while with this skip list implementation it would take 110 steps

Reply:
> I don't really understand it myself but I was hoping it was just me 😀 I guess not.

Ha yeah no stupid questions mate, and besides sometimes the explanation is not described well enough and we should aim to change that.

@Aprogiena thanks for the more detailed description 👍 
Your description should probably go in the method headers.

Reply:
The description says 
>max 110 steps to go back up to 2^18 blocks

This is just to traverse back 2^18 to genesis but it could mean more steps to hit the target requested block right?

Perhaps this should also described in the comments.

Reply:
that comment is just telling you how many steps you should expect
if the chain is shorter than 2^18, then there should not be more than 110 steps
if the chain is longer, there will be more steps, this number is just an example what to expect
it is O(log(N)) time complexity alg